src/Bridge/Doctrine/Purger/ManagerRegistryPurger.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Fidry\AliceDataFixtures package.
+ *
+ * (c) Théo FIDRY <theo.fidry@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Fidry\AliceDataFixtures\Bridge\Doctrine\Persister;
+
+use function array_map;
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\Common\Persistence\ObjectManager;
+use Fidry\AliceDataFixtures\Bridge\Doctrine\Purger\ObjectManagerPurger;
+use Fidry\AliceDataFixtures\Persistence\PurgeMode;
+use Fidry\AliceDataFixtures\Persistence\PurgerFactoryInterface;
+use Fidry\AliceDataFixtures\Persistence\PurgerInterface;
+use InvalidArgumentException;
+use Nelmio\Alice\IsAServiceTrait;
+
+/**
+ * @final
+ */
+/* final */ class ManagerRegistryPurger implements PurgerInterface, PurgerFactoryInterface
+{
+    use IsAServiceTrait;
+
+    private $registry;
+    private $purgeMode;
+
+    /**
+     * @var PurgerInterface[]
+     */
+    private $purgers = [];
+
+    public function __construct(ManagerRegistry $registry, PurgeMode $purgeMode = null)
+    {
+        $this->registry = $registry;
+
+        $this->purgers = array_map(
+            function (ObjectManager $manager) use ($purgeMode): PurgerInterface {
+                return new ObjectManagerPurger($manager, $purgeMode);
+            },
+            $registry->getManagers()
+        );
```

Comment:
This part here makes the assumption that data from all managers (and hence databases) should be purged every time.

IMHO this seems counter intuitive when running `bin/console hautelook:fixtures:load --manager=foo`. I certainly wouldn't expect the database of manager `bar` be also purged.

Looking at the problem from a wider perspective, perhaps it would be more sensible to remove the `--manager` option and instead create a purger for only those managers which contain at least a single entity with fixtures defined for it?

Reply:
If that could be done I guess it would be less potential side-effects indeed...